Automattic/jetpack
# Blocks: remove retired Revue block and add standardized way to display fallback block content

## Proposed changes:

The block was retired in #27961, when the service shut down back in December of last year. It's now been 6 months, we can consequently safely remove the block altogether.

As part of this PR, I'm introducing 2 new things:

1. A list of deprecated blocks, where we'll add any other blocks that we stop supporting in the future.
2. A function hooked into the `render_block` hook to display a custom message if someone happens to still be using a deprecated block. That custom message will only be displayed for editors+; site visitors will see nothing.

This setup will allow us to keep the frontend clean of any old code that may not be functional anymore. In the editor, however, we'll let WP Core handle things and display the usual message showing the block is not supported:

<img width="750" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/d2a5d83b-eba0-4c4b-980d-6b0deb540c1a">

Here is how things will look like on the frontend:

**Logged out**

<img width="907" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/15e1250a-074f-473a-8043-509a2f6df725">

**Logged in**

<img width="908" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/61b3690a-60e7-4802-b0e8-80920a216aff">


### Other information:


## Jetpack product discussion

Internal reference: 1202723222637401-as-1203700397893924

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

2 main things to test here:

1. you should not find any more references to Revue in our codebase, and should not be able to insert a new Revue block on your site.
2. Existing Revue blocks have returned a warning in the past 6 months, now they'll display in 3 different ways:
    1. In the editor, you'll see an error if you still haven't removed them.
    2. On the frontend when logged in as an editor+, you'll see a message inviting you to remove the block.
    3. On the frontend when logged out, you wil see nothing.

To test this, you could try adding a Revue block to your site by pasting the following into the code editor view of the block editor:

```

<div class="wp-block-jetpack-revue">

<a class="wp-block-jetpack-revue__fallback" href="https://www.getrevue.co/profile/jeherve">https://www.getrevue.co/profile/jeherve</a></div>

```
